src/main/java/com/artipie/scripting/ScriptRunner.java

```diff
@@ -0,0 +1,120 @@
+import org.quartz.impl.StdSchedulerFactory;
+
+/**
+ * Script runner.
+ * Job for running script in quartz
+ *
+ * @since 0.30
+ */
+public final class ScriptRunner implements Job {
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        final Settings settings = (Settings) context.getJobDetail().getJobDataMap().get("settings");
+        final Key key = new Key.From(context.getJobDetail().getJobDataMap().getString("key"));
+        final BlockingStorage storage = new BlockingStorage(settings.configStorage());
+        if (settings == null || key == null || storage == null) {
+            this.stopJob(context);
+            return;
+        }
+        if (storage.exists(key)) {
+            extension(key.toString())
+                .flatMap(ext -> script(ext, new String(storage.value(key))))
+                .map(
+                    script -> {
+                        Optional<Script.Result> result;
+                        try {
+                            final Map<String, Object> vars = new HashMap<>();
+                            vars.put("_settings", settings);
+                            vars.put("_repositories", new RepositoriesFromStorage(settings));
+                            result = Optional.of(script.call(vars));
+                        } catch (final ScriptException exc) {
+                            Logger.error(
+                                ScriptRunner.class,
+                                "Execution error in script %s %[exception]s",
+                                key.toString(),
+                                exc
+                            );
+                            result = Optional.empty();
+                        }
+                        return result;
+                    }
+                );
+        } else {
+            Logger.warn(ScriptRunner.class, "Cannot find script %s", key.toString());
+        }
+    }
+
+    /**
+     * Stops the job and logs error.
+     * @param context Job context
+     */
+    private void stopJob(final JobExecutionContext context) {
+        final JobKey key = context.getJobDetail().getKey();
+        try {
+            Logger.error(this, String.format("Force stopping job %s...", key));
+            new StdSchedulerFactory().getScheduler().deleteJob(key);
+            Logger.error(this, String.format("Job %s stopped.", key));
+        } catch (final SchedulerException error) {
+            Logger.error(this, String.format("Error while stopping job %s", key));
+            throw new ArtipieException(error);
+        }
+    }
+
+    /**
+     * Create instance of Script by script-file extension and script code.
+     * @param ext Extension of script-file.
+     * @param script Script code.
+     * @return Script instance
+     */
+    private static Optional<Script> script(final String ext, final String script) {
+        final Map<String, Script.ScriptType> map = Map.of(
```

Comment:
I'd suggest use loop via enum values instead of map to get script type. Thus the code will work just fine and no changes will be required if we add another script type.